BO_training/mixed_test_func/scallop_casmopolitan_yaml.py

```diff
@@ -321,3 +323,170 @@ def coordinate_ascent_warmup(Function,fold_step=1,iterations=20):
             old_fold_step = old_fold_step/2.0
         temp_fold_step = old_fold_step
     return (ca_X,ca_Y)
+
+def coordinate_ascent_warmup_yaml(f, max_iters=60,path='',num_threads = 1):
```

Comment:
This function is much too long and much too deep (too many ifs in while in while). It is very difficult to read and maintain as is.

General readability guidelines:
* A function should fit in 1 computer screen. So you can look at it at once. Longer functions require that you remember too much things.
* The indentation level should not go beyond 3. You go to 7. It is then hard to understand the logic of the code, and editing the code is very error prone.

Please refactor this function into multiple independent functions.